hibernate-jpa/src/main/java/io/micronaut/configuration/hibernate/jpa/EntityManagerFactoryBean.java

```diff
@@ -107,11 +105,8 @@ protected StandardServiceRegistry hibernateStandardServiceRegistry(
 
         Map<String, Object> additionalSettings = new LinkedHashMap<>();
         additionalSettings.put(AvailableSettings.DATASOURCE, dataSource);
-        Class<?> sessionContextClass = ClassUtils.forName("org.springframework.orm.hibernate5"
-                                                        + ".SpringSessionContext", EntityManagerFactoryBean.class.getClassLoader())
-                .orElse(MicronautSessionContext.class);
-        additionalSettings.put(
-                AvailableSettings.CURRENT_SESSION_CONTEXT_CLASS, sessionContextClass.getName());
+        additionalSettings.put(AvailableSettings.CURRENT_SESSION_CONTEXT_CLASS,
+                beanLocator.getBean(HibernateCurrentSessionContextClassProvider.class).get().getName());
```

Comment:
This may need to use `containsBean` with a fallback for the case where it is missing?

Reply:
There is a default `MicronautHibernateCurrentSessionContextClassProvider` or I can remove it and return `MicronautSessionContext` if no provider exists.

Reply:
Done